.github/files/test-wpcom-filename-restrictions.sh

```diff
@@ -0,0 +1,184 @@
+#!/bin/bash
+
+set -eo pipefail
+
+source "$GITHUB_WORKSPACE/trunk/.github/files/gh-funcs.sh"
+
+# Based on Automattic/pre-receive-hooks/blob/8bf13a23/common/050-stop-underscores.sh
+IGNORE_UNDERSCORE_RULE_FOR='bin/wp-cli|bin/atoum|bin/wp-cli-wpcom|wp-includes/random_compat|wp-includes/sodium_compat|wp-content/plugins/glotpress|.phabricator-linter|wp-content/lib/customer-lists/|wp-content/lib/feature-store/vendor/|wp-content/lib/nosara/ThriftSQL.src/ThriftGenerated/|wp-content/mu-plugins/jetpack-plugin/vendor/|wp-content/lib/aws/vendor/|wp-content/mu-plugins/crowdsignal-forms/vendor/|wp-content/plugins/woocommerce/|wp-content/plugins/amp-2.0/|wp-content/plugins/woocommerce-payments/|wp-content/plugins/woocommerce-subscriptions/|wp-content/plugins/p2(-wpcom)?|wp-content/lib/google/|wp-content/lib/tus-php/|wp-content/vip-plugins/facebook-instant-articles-3.2/|wp-content/vip-plugins/facebook-instant-articles-4.0/|wp-content/mu-plugins/jetpack-packages/|wp-content/plugins/woo-gutenberg-products-block/vendor/|/autoload_|/vendor/composer/'
+function check_underscores {
+	local FILE="$1"
+	if echo "$PREFIX/$FILE" |
+		grep '\.php$' |
+		egrep -v "${IGNORE_UNDERSCORE_RULE_FOR}" |
+		grep _ |
+		rev |
+		cut -d'/' -f 1 |
+		rev |
+		grep _ &>/dev/null
```

Comment:
The existing flow seems better suited to handle many filenames, but we're passing them one by one.

I'm a bit confused by this, though it seems to be what is used in the source script:

* It's grepping for `_` twice
* It's using both `grep` and `egrep` (why not `grep -E`?)
* It seems to be using `rev | cut | rev` when I think it could just use basename?
* There's no way to exit early (e.g. if it isn't a PHP file or matches the ignore, it continues processing).

Any advantage vs. something like this, which to me is way more readable?

```
if [[ "$PREFIX/$FILE" == *.php && "$(basename $FILE)" =~ && ! "$PREFIX/$FILE" =~ $IGNORE_UNDERSCORE_RULE_FOR ]]
```

Reply:
I wanted to keep closer to what's in the wpcom check, so if they change things in the check we can more easily compare. This is pretty much a copy-paste.